### PR TITLE
Fix incorrect Gradle root build directory location

### LINKED ISSUE #11971: Gradle 6.1+ generates an empty .gradle and gradle directories on each execution in the current working folder
### Expected Behavior

Gradle should not generate dummy directories on its own.

### Current Behavior

Gradle generates `gradle` in the current working directory.
Sometimes it creates `.gradle` :((

It looks like https://github.com/gradle/gradle/commit/438e4f99dc17024e74d8e0f3c6ab2719c364e159#diff-cffca3a015bacf898383932f0abada06R40-R43 results in creating `gradle` directory on each execution (e.g. even if settings.gradle.kts throws an exception and includes 0 projects).

The same issue can be reproduced in gradle-6.2-20200121230035+0000-all.zip

@melix , is it something you know?

### Steps to Reproduce 

```
1. create_gradle_6.1_project
$ mkdir tmp && cd tmp # create a temp directory and cd inside
$ ls -la # ensure the directory is empty
.
..
$ ../gradlew qwerty # <-- "../" is important !!
FAILURE: Build failed with an exception.

* What went wrong:
Task 'qwerty' not found in root project 'test'.
```
^^ task failure is expected

Note: new folders appear after Gradle failure:

```
$ ls -la 
.
..
.gradle # <-- watch this
gradle # <-- and this
```

### PR DESCRIPTION
**Pull request by melix**
### Context

Fixes #11971

### CODE REVIEW

**Comment by melix** [comment]
I also have prepared a branch merging into `master` because it's not trivial merge.

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java [hunk 1]
```diff
@@ -334,8 +337,10 @@ RepositoryBlacklister createRepositoryBlacklister() {
         return new ConnectionFailureRepositoryBlacklister();
     }
 
-    StartParameterResolutionOverride createStartParameterResolutionOverride(StartParameter startParameter) {
-        return new StartParameterResolutionOverride(startParameter);
+    StartParameterResolutionOverride createStartParameterResolutionOverride(StartParameter startParameter, BuildLayoutFactory buildLayoutFactory) {
```
Comment (big-guy) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java]: Can you use `BuildLayout` here?  It should be a build scope service created in this same way.

**Comment by melix** [comment]
Verification is not public in 6.1 so it doesn't matter really.

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java (hunk 1 above)
Comment (melix) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildScopeServices.java]: Nope, `BuildLayout` is not available at this stage.

**Pull request merged**